src/lambda/expression.rs

```diff
@@ -144,7 +151,13 @@ impl Expression {
               }
               if ctx.req_ctx.server.get_enable_cache_control() && res.status.is_success() {
                 if let Some(max_age) = max_age(&res) {
-                  ctx.req_ctx.set_min_max_age(max_age.as_secs());
+                  ctx.req_ctx.set_min_max_age(max_age.as_secs() as i64);
+                }
+                if let Some(policy) = cache_policy(&res) {
+                  ctx.req_ctx.set_cache_visibility(&policy.cachability);
+                  if Some(Cachability::NoCache) == policy.cachability {
+                    ctx.req_ctx.set_min_max_age(-1);
+                  }
```

Comment:
looks like we have a duplicated code here, could you extract this to a helper function?

Reply:
Refactored to a new method `set_cache_control`.